src/compiler/transformers/classFields.ts

```diff
@@ -7,95 +7,72 @@ namespace ts {
          */
         ClassAliases = 1 << 0,
     }
-
-    const enum PrivateIdentifierPlacement {
-        InstanceField,
-        InstanceMethod,
-        InstanceGetterOnly,
-        InstanceSetterOnly,
-        InstanceGetterAndSetter,
-        StaticField,
-        StaticMethod,
-        StaticGetterOnly,
-        StaticSetterOnly,
-        StaticGetterAndSetter,
-    }
-
-    type PrivateIdentifierInfo =
-        | PrivateIdentifierStaticField
-        | PrivateIdentifierStaticMethod
-        | PrivateIdentifierStaticGetterOnly
-        | PrivateIdentifierStaticSetterOnly
-        | PrivateIdentifierStaticGetterAndSetterOnly
-        | PrivateIdentifierInstanceField
-        | PrivateIdentifierInstanceMethod
-        | PrivateIdentifierInstanceGetterOnly
-        | PrivateIdentifierInstanceSetterOnly
-        | PrivateIdentifierInstanceGetterAndSetter;
-
-    interface PrivateIdentifierStaticField {
-        placement: PrivateIdentifierPlacement.StaticField;
-        variableName: Identifier;
-        classConstructor: Identifier;
+    export const enum PrivateIdentifierKind {
+        Field = "f",
+        Method = "m",
+        Accessor = "a"
     }
-    interface PrivateIdentifierStaticMethod {
-        placement: PrivateIdentifierPlacement.StaticMethod;
-        functionName: Identifier;
-        classConstructor: Identifier;
-    }
-    interface PrivateIdentifierStaticGetterOnly {
-        placement: PrivateIdentifierPlacement.StaticGetterOnly;
-        getterName: Identifier;
-        classConstructor: Identifier;
-    }
-    interface PrivateIdentifierStaticSetterOnly {
-        placement: PrivateIdentifierPlacement.StaticSetterOnly;
-        setterName: Identifier;
-        classConstructor: Identifier;
-    }
-    interface PrivateIdentifierStaticGetterAndSetterOnly {
-        placement: PrivateIdentifierPlacement.StaticGetterAndSetter;
-        getterName: Identifier;
-        setterName: Identifier;
-        classConstructor: Identifier;
+    interface PrivateIdentifierInfoBase {
+        /**
+         * State can contain:
+         *  - For instance field: The WeakMap that will be the storage for teh field.
+         *  - For instance methods or accessors: The WeakSet that will be used.
+         *  - For static members: The constructor that will be used for brand checking.
+         */
+        state: Identifier;

```

Comment:
nit: Do you think `brandCheckIdentifier` would be a better name? I know it's more than that for instance fields but `state` is a bit vague. No strong opinions though.

Reply:
I actually like `brandCheckIdentifier`.. the WeakMap is a bit more for instance fields .. but it still is the brand check as well.